library/iFixit/Matryoshka/Memcached.php

```diff
@@ -36,7 +36,12 @@ protected function __construct(\Memcached $memcached) {
    }
 
    public function set($key, $value, $expiration = 0) {
-      return $this->memcached->set($key, $value, $expiration);
+      $setValue = $this->memcached->set($key, $value, $expiration);
+
+      if ($this->memcached->getResultCode() === \Memcached::RES_E2BIG) {
+         throw new ValueTooLargeException();
+      }
```

Comment:
After doing some digging I think you will want to mimic https://github.com/iFixit/Matryoshka/pull/26/ which decided to report any non-expected state.
```suggestion
      if ($resultCode !== \Memcached::RES_SUCCESS) {
         $errorMessage = $this->memcached->getResultMessage();
         throw new \MemcachedException($errorMessage, $resultCode);
      }
```

You may need ensure `Memcached::MISS` is not expected when calling set. (I don't expect that to make sense, but who knows)

Reply:
Set should just return true or false.